cmd/kuberhealthy/crd.go

```diff
@@ -64,7 +65,7 @@ func sanitizeResourceName(c string) string {
 }
 
 // ensureStateResourceExists checks for the existence of the specified resource and creates it if it does not exist
-func ensureStateResourceExists(checkName string, checkNamespace string) error {
+func ensureStateResourceExists(checkName string, checkNamespace string, k *Kuberhealthy) error {
```

Comment:
Could we refactor this function so that we pass in the owner ref instead of the `Kuberhealthy` instance?  This would shorten the code here and make it assume less things.  This would of course mean finding all the calls to this func and augmenting them with the proper ownerRef data.  That owner ref could probably be a global in the invoking Kuberhealthy program, though - which is faster than querying it every time.

What do you think?

Reply:
we need Kuberhealthy instance to run 
```
checker, err := k.getCheck(checkName, checkNamespace)
if err != nil {
	return err
}
// Get ownerReference for the kuberhealthy khstate
ownerRef, err := checker.GenOwnerRef()
if err != nil {
	return errors.New("Failed to getOwnerReference for khstate: " + initialState.Name + ", err: " + err.Error())
}
```
but i think make `ensureStateResourceExists ` to a method for Kuberhealthy instance is a good idea.
```
func (k *Kuberhealthy) ensureStateResourceExists(checkName string, checkNamespace string) error {
.......
}
```
i'll fix it

Reply:
I think a method works here. Thank you.